### PR TITLE
[fix](remote scan) Fix remote scan paralism

### PR DESCRIPTION
**Pull request by zhiqiang-hhhh**
Introduced by https://github.com/apache/doris/pull/41273.

The modification of _max_scan_thread_num is a long story. So, the modification i will describe only contains three timestamp. 1. Before https://github.com/apache/doris/pull/41273; 2. After https://github.com/apache/doris/pull/41273; 3. After this pr.

We have two dimension, so i will use a table. The value of the table is the _max_scan_thread_num when queries running **serially**.

C refers to num of cpu cores.

1. Before 41273

|Engine|OLAPScanOperator|FileScanOperator|
|---|---|---|
|PipelineX|min(2C, num_of_scanner)|4|
|Old-Pipeline|4|4|
|Non-Pipeline|4|4|

2. After 41273

|Engine|OLAPScanOperator|FileScanOperator|
|---|---|---|
|PipelineX|min(2C, num_of_scanner)|min(2C, num_of_scanner)|
|Old-Pipeline|4|4|
|Non-Pipeline|4|4|

3. After this pr

|Engine|OLAPScanOperator|FileScanOperator|
|---|---|---|
|PipelineX|min(2C, num_of_scanner)|4|
|Old-Pipeline|4|4|
|Non-Pipeline|4|4|

### CODE REVIEW

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

**Comment by zhiqiang-hhhh** [comment]
run buildall

File: be/src/vec/exec/scan/scanner_context.h [hunk 1]
```diff
@@ -234,6 +235,7 @@ class ScannerContext : public std::enable_shared_from_this<ScannerContext>,
     RuntimeProfile::Counter* _scale_up_scanners_counter = nullptr;
     QueryThreadContext _query_thread_context;
     bool _ignore_data_distribution = false;
+    bool _is_file_scan_operator;
```
Comment (yiguolei) [inline be/src/vec/exec/scan/scanner_context.h:238]: should set default value

**Pull request merged**